document-generator-company-report/src/main/java/uk/gov/companieshouse/document/generator/company/report/handler/CompanyReportDataHandler.java

```diff
@@ -43,32 +46,49 @@
 @Component
 public class CompanyReportDataHandler {
 
-    @Autowired
     private CompanyService companyService;
 
-    @Autowired
     private PscsService pscsService;
 
-    @Autowired
     private OfficerService officerService;
 
-    @Autowired
     private UkEstablishmentService ukEstablishmentService;
 
-    @Autowired
     private RecentFilingHistoryService recentFilingHistoryService;
 
-    @Autowired
     private CompanyReportMapper companyReportMapper;
 
-    @Autowired
     private StatementsService statementsService;
 
+    private InsolvencyService insolvencyService;
+
+    @Autowired
+    public CompanyReportDataHandler(CompanyService companyService,
+                                    PscsService pscsService,
+                                    OfficerService officerService,
+                                    UkEstablishmentService ukEstablishmentService,
+                                    RecentFilingHistoryService recentFilingHistoryService,
+                                    CompanyReportMapper companyReportMapper,
+                                    StatementsService statementsService,
+                                    InsolvencyService insolvencyService) {
+
+        this.companyService = companyService;
+        this.pscsService = pscsService;
+        this.officerService = officerService;
+        this.ukEstablishmentService = ukEstablishmentService;
+        this.recentFilingHistoryService = recentFilingHistoryService;
+        this.companyReportMapper = companyReportMapper;
+        this.statementsService = statementsService;
+        this.insolvencyService = insolvencyService;
```

Comment:
Sonar's complaining about having greater than 7 parameters in the constructor. This may need to be revisited once we have all aspects of the report done

Reply:
refactoring being done on separate branch